pylmod/membership.py

```diff
@@ -19,14 +19,111 @@ def __init__(
+
+        Args:
+            uuid (str): optional uuid. defaults to self.cuuid
+
+        Raises:
+            PyLmodUnexpectedData: No data was returned.
+            requests.RequestException: Exception connection error
+
+        Returns:
+            dict: group json
+
+        """
+        if uuid is None:
+            uuid = self.uuid
+        group_data = self.get('group', params={'uuid': uuid})
+        return group_data
+
+    def get_group_id(self, uuid=None):
+        """Get group id based on uuid.
+
+        Args:
+            uuid (str): optional uuid. defaults to self.cuuid
+
+        Raises:
+            PyLmodUnexpectedData: No group data was returned.
+            requests.RequestException: Exception connection error
+
+        Returns:
+            int: numeric group id
+
+        """
+        group_data = self.get_group(uuid)
+        try:
+            return group_data['response']['docs'][0]['id']
+        except (KeyError, IndexError):
+            failure_message = ('Error in get_group response data - '
+                               'got {0}'.format(group_data))
+            log.exception(failure_message)
+            raise PyLmodUnexpectedData(failure_message)
+
+    def get_membership(self, uuid=None):
+        """Get membership data based on uuid.
+
+        Args:
+            uuid (str): optional uuid. defaults to self.cuuid
+
+        Raises:
+            PyLmodUnexpectedData: No data was returned.
+            requests.RequestException: Exception connection error
+
+        Returns:
+            dict: membership json
+
+        """
+        group_id = self.get_group_id(uuid=uuid)
+        uri = 'group/{group_id}/member'
+        mbr_data = self.get(uri.format(group_id=group_id), params=None)
+        return mbr_data
+
+    def email_has_role(self, email, role_name, uuid=None):
+        """Determine if an email is associated with a role.
+
+        Args:
+            email (str): user email
+            role_name (str): user role
+            uuid (str): optional uuid. defaults to self.cuuid
+
+        Raises:
+            PyLmodUnexpectedData: Unexpected data was returned.
+            requests.RequestException: Exception connection error
+
+        Returns:
+            bool: True or False if email has role_name
+
+        """
+        mbr_data = self.get_membership(uuid=uuid)
+        docs = []
+        try:
+            docs = mbr_data['response']['docs']
+        except KeyError:
+            failure_message = ('KeyError in membership data - '
+                               'got {0}'.format(mbr_data))
+            log.exception(failure_message)
+            raise PyLmodUnexpectedData(failure_message)
+        if len(docs) == 0:
+            return False
+        has_role = any(
+            (x.get('email') == email and x.get('roleType') == role_name)
+            for x in docs
+        )
+        if has_role:
+            return True
+        return False
 
     def get_course_id(self, course_uuid):
```

Comment:
I'd expect get_course_id() and get_group_id() to have the same signature.  I don't care which you change, but it would make the code more homogeneous.